packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -267,6 +267,142 @@ class _UntilTextBoundary extends TextBoundary {
   }
 }
 
+/// A class that describes how textual contents should be scaled for better
+/// readability.
+///
+/// The [scale] function computes the scaled font size given the original
+/// unscaled font size specified by app developers.
+///
+/// The [==] operator defines the equality of 2 [TextScaler]s, which the
+/// framework uses to determine whether text widgets should rebuild when their
+/// [TextScaler] changes. Consider overridding the [==] operator if applicable
+/// to avoid unnecessary rebuilds.
+@immutable
+abstract class TextScaler {
+  /// Creates a TextScaler.
+  const TextScaler();
+
+  /// Creates a proportional [TextScaler] that scales the incoming font size by
+  /// multiplying it with the given `textScaleFactor`.
+  const factory TextScaler.linear(double textScaleFactor) = _LinearTextScaler;
+
+  /// A [TextScaler] that doesn't scale the ipnut font size.
+  ///
+  /// This is equivalent to `TextScaler.linear(1.0)`, the [TextScaler.scale]
+  /// implementation always returns the input font size as-is.
+  static const TextScaler noScaling = _LinearTextScaler(1.0);
+
+  /// Computes the scaled font size (in logical pixels) with the given unscaled
+  /// `fontSize` (in logical pixels).
+  ///
+  /// The input `fontSize` must be finite and non-negative.
+  ///
+  /// When given the same `fontSize` input, this method returns the same value.
+  /// The output of a larger input `fontSize` is typically larger than that of a
+  /// smaller input, but on unusual occasions they may produce the same output.
+  /// For example, some platforms use single-precision floats to represent font
+  /// sizes, as a result of truncation two different unscaled font sizes can be
+  /// scaled to the same value.
+  double scale(double fontSize);
+
+  /// The estimated number of font pixels for each logical pixel.
+  ///
+  /// The value of this property is only an estimate, so it may not reflect the
+  /// exact text scaling strategy this [TextScaler] represents, especially when
+  /// this [TextScaler] is not linear. Consider using [TextScaler.scale] instead.
+  /// This property exists solely for backward compatibility purposes, such that
```

Comment:
If so, we should mark this `@Deprecated`

Reply:
otherwise, we should be more confident about the usage of this property. Right now the document sounds like this will probably go away in the future.